roles/swlib/tasks/gcstransfer.yml

```diff
@@ -13,137 +13,53 @@
 # limitations under the License.
 
 ---
-# files with alt_name are stored as name for simplicity
-- name: gcstransfer | Transfer GI software from GCS to target instance
-  shell: |
-    set -o pipefail
-    {% for f in item.files %}
-    if gcloud storage ls gs://{{ swlib_mount_src }}/{{ f.name }} >/dev/null 2>&1; then
-      f_name="{{ f.name }}"
-    else
-      f_name="{{ f.alt_name | default('x') }}"
-    fi
-    if ! ssh -i {{ ansible_ssh_private_key_file }} {{ ansible_ssh_extra_args }} {{ ansible_ssh_user }}@{{ ansible_ssh_host }} \
-      "unzip -l {{ swlib_path }}/{{ f.name }}"
-      then
-        gcloud storage cat gs://"{{ swlib_mount_src }}"/"${f_name}" | ssh -i {{ ansible_ssh_private_key_file }} {{ ansible_ssh_extra_args }} \
-          {{ ansible_ssh_user }}@{{ ansible_ssh_host }} "cat > {{ swlib_path }}/{{ f.name }}"
-    fi
-    {% endfor %}
-  register: shell_result
-  args:
-    executable: /bin/bash
-  with_items:
-    - "{{ gi_software }}"
-    - "{{ gi_interim_patches }}"
-  when:
-    - gi_install
-    - item.version == oracle_ver
-    - patch_only is not defined
-  changed_when: "'Date    Time    Name' not in shell_result.stdout"
-  delegate_to: 127.0.0.1
-
-- name: gcstransfer | Transfer GI patches from GCS to target instance
-  shell: |
-    set -o pipefail
-    if ! ssh -i {{ ansible_ssh_private_key_file }} {{ ansible_ssh_extra_args }} {{ ansible_ssh_user }}@{{ ansible_ssh_host }} \
-      "unzip -l {{ swlib_path }}/{{ item.patchfile }}"
-      then
-        gcloud storage cat gs://{{ swlib_mount_src }}/{{ item.patchfile }} | ssh -i {{ ansible_ssh_private_key_file }} {{ ansible_ssh_extra_args }} \
-          {{ ansible_ssh_user }}@{{ ansible_ssh_host }} "cat > {{ swlib_path }}/{{ item.patchfile }}"
-    fi
-  register: shell_result
-  args:
-    executable: /bin/bash
-  with_items: "{{ gi_patches }}"
-  when:
-    - gi_install
-    - item.release == oracle_rel
-  changed_when: "'Date    Time    Name' not in shell_result.stdout"
-  delegate_to: 127.0.0.1
+- name: gcstransfer | Set the Ansible ssh command for use in subsequent file tasks
+  set_fact:
+    ssh_cmd: "ssh -i {{ ansible_ssh_private_key_file }} {{ ansible_ssh_extra_args }} {{ ansible_ssh_user }}@{{ ansible_ssh_host }}"
```

Comment:
Would we get a benefit by setting this globally?  It seems like the kind of thing we might use in other playbooks too.

Reply:
In that case let's leave as-is;  we can refactor later.